herd/test_herd.ml

```diff
@@ -66,6 +66,84 @@ module Make(A:Arch_herd.S) =
        A.prop, A.location, A.RLocSet.t,A.FaultAtomSet.t) t
 
 
+    let add_output_address acc = function
+      | OutputAddress.PTE s
+      | OutputAddress.PHY s -> StringSet.add s acc
+
+    let add_constant_symbols acc c =
+      let open Constant in
+      match c with
+      | Symbolic sym ->
+          (match sym with
+          | Virtual { name = base; _ } ->
+              if Symbol.is_data base then
+                StringSet.add (Symbol.pp base) acc
+              else acc
+          | Physical (name, _) -> StringSet.add name acc
+          | TagAddr (_, name, _) -> StringSet.add name acc
+          | System (_, name) -> StringSet.add name acc)
+      | PteVal p ->
+          let open ParsedPteVal in
+          (match p.p_oa with
+          | None -> acc
+          | Some oa -> add_output_address acc oa)
+      | AddrReg a ->
+          let open ParsedAddrReg in
+          (match a.p_oa with
+          | None -> acc
+          | Some oa -> add_output_address acc oa)
+      | _ -> acc
+
+    let add_location_symbols loc acc =
+      match loc with
+      | MiscParser.Location_global c -> add_constant_symbols acc c
+      | MiscParser.Location_sreg name -> StringSet.add name acc
+      | MiscParser.Location_reg _ -> acc
+
+    let add_rloc_symbols rloc acc =
+      ConstrGen.fold_rloc add_location_symbols rloc acc
+
+    let add_init_symbols init acc =
+      List.fold_left
+        (fun acc (loc,(_,v)) ->
+          let acc = add_location_symbols loc acc in
+          add_constant_symbols acc v)
+        acc init
+
+    let add_fault_symbols (_,loc_opt,_) acc =
+      match loc_opt with
+      | None -> acc
+      | Some c -> add_constant_symbols acc c
+
+    let add_postcondition_symbols constr acc =
```

Comment:
Should/can we move this to `lib/constrGen.ml` and all the functions it depends on?